packages/liveblocks-core/src/LiveObject.ts

```diff
@@ -515,6 +517,7 @@ export class LiveObject<O extends LsonObject> extends AbstractCrdt {
    * @param patch The object used to overrides properties
    */
   update(patch: Partial<O>): void {
+    this._pool?.assertStorageIsWritable();
     if (this._pool === undefined || this._id === undefined) {
```

Comment:
Not super sure what this `if`-block is handling. Could the logic in here still be writing to the pool through related Live objects, indirectly?

Reply:
To be perfectly frank I am having trouble understanding as well 😅 

My rough understanding is that it goes through on this if it's a local / offline LiveObject that was set? Since if `_id` is `undefined` it's not yet attached to the storage and if `_pool` is `undefined` then you're offline, regardless, the server would then provide the source of truth.

@GuillaumeSalles would you be able to shine some light on this by any chance?